src/plugins/data/common/opensearch_query/opensearch_query/filter_matches_index.ts

```diff
@@ -31,14 +31,14 @@
 import { IIndexPattern, IFieldType } from '../../index_patterns';
 import { Filter } from '../filters';
 
-/*
- * TODO: We should base this on something better than `filter.meta.key`. We should probably modify
- * this to check if `filter.meta.index` matches `indexPattern.id` instead, but that's a breaking
- * change.
- */
```

Comment:
Let's make an issue that captures the uncertainty we still have here, where we can discuss the other underlying issues, including lack of validation on the custom filter inputs.